src/app/GitCommands/CommitMessageManager.cs

```diff
@@ -230,9 +248,8 @@ private async Task<string> ReadFileAsync(string filePath, string errorTitle, Enc
             }
             catch (Exception ex) when (ex is not (OperationCanceledException or ObjectDisposedException))
             {
-                await _owner.SwitchToMainThreadAsync();
-                MessageBox.Show(_owner, string.Format(CannotAccessFile, ex.Message, filePath), errorTitle, MessageBoxButtons.OK, MessageBoxIcon.Exclamation);
-                return string.Empty;
+                string message = string.Format(CannotAccessFile, ex.Message, filePath);
+                throw new CannotAccessFileException(message, ex);
```

Comment:
Either
```suggestion
                ThreadHelper.FileAndForget(() => throw new UserNotificationException(new CannotAccessFileException(message, ex));
```
or rather without wrapping in yet another exception
````suggestion
                ThreadHelper.FileAndForget(() => throw new UserNotificationException(message, ex));
```

Reply:
Looks like winforms tricks. Why unit tests must know AppDomain or Application. I want to mock it all.
The service class must not be independent of the concrete realization. I don't understand why your team would use WinForms tricks instead of a well-known pattern.